ComplementaryScripts/modelCuration/AddCofactorIon.m

```diff
@@ -0,0 +1,216 @@
+        'reversible',newrxn.Rev(i,1),...
+        'geneRule',newrxn.GPR{i},...
+        'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
+    if isempty(rxnIndex)
+        rxnIndex = strcmp(model.rxns,['r_' newID]);
+    end
+    % Add rxn annotation:
+    model.rxnNames{rxnIndex}      = newrxn.rxnNames{i};
+    model.rxnECNumbers(rxnIndex)  = newrxn.rxnECNumbers(i);
+    model.rxnKEGGID(rxnIndex)     = newrxn.rxnKEGGID(i);
+    model.rxnMetaNetXID(rxnIndex) = newrxn.rxnMetaNetXID(i);
+    model.rxnConfidenceScores(rxnIndex) = 1;   %reactions without gene but needed for modelling
+    model.rxnNotes{rxnIndex} = 'NOTES:added for BiomassUpdate';
+end
+
+% add gene standard name for new genes
+fid = fopen('../../ComplementaryData/databases/SGDgeneNames.tsv');
+yeast_gene_annotation = textscan(fid,'%s %s','Delimiter','\t','HeaderLines',1);
+fclose(fid);
+
+geneIndex = zeros(1,1);
+for i = 1: length(model.genes)
+    geneIndex = strcmp(yeast_gene_annotation{1}, model.genes{i});
+    if sum(geneIndex) == 1 && ~isempty(yeast_gene_annotation{2}{geneIndex})
+        model.geneNames{i} = yeast_gene_annotation{2}{geneIndex};
+    else
+        model.geneNames{i} = model.genes{i};
+    end
+end
+
+% Add protein name for genes
+for i = 1:length(model.genes)
+    model.proteins{i} = strcat('COBRAProtein',num2str(i));
+end
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Introduce two PseudoRxn for cofactors and ions
+group = unique(data.groups);
+for i = 1:length(group)
+%get new metID for pseudo mets :ion and cofactor
+    metID  = ['s_' getNewIndex(model.mets) '[c]'];
+    model  = addMetabolite(model,metID,'metName',[group{i} ' [cytoplasm]']);
+    [~,index] = ismember(metID,model.mets);
+    model.metNotes{index}      = 'NOTES: added for BiomassUpdate';
+
+% Get coefficients for pseudo rxns: ion pseudoreaction, cofactor pseudoreaction
+    newps.rxnName = {[group{i},' pseudoreaction']};
+    index = strcmp(group{i},data.groups);
+    newps.metIDs = data.mets(index);
+    newps.metcoef = data.abundances(index)*-1;
+
+% introduce two pseudo mets ion [cytoplasm], cofactor [cytoplasm] into the reaction
+    newps.metIDs = [newps.metIDs;metID];
+    newps.metcoef = [newps.metcoef;1];
+
+    rxnID = ['r_' getNewIndex(model.rxns)];
+    [model,rxnIndex] = addReaction(model, rxnID,...
+                       'reactionName', char(newps.rxnName),...
+                       'metaboliteList',newps.metIDs,...
+                       'stoichCoeffList',newps.metcoef,...
+                       'reversible',false,...
+                       'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
```

Comment:
Same as before for lines 191-192: `'CheckEnergyProduction' is not found in the current folder or on the MATLAB path, but exists in: C:\Users\bensan\Documents\repos\yeast-GEM\ComplementaryScripts\modelCuration`